### PR TITLE
Bazel version update to 6.2

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

**Comment by napetrov** [comment]
to simplify things - this would require switch to bazilesk in publicCI as well as in internal CI

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 1]
```diff
@@ -33,15 +32,15 @@ TEST(column_accessor_test, can_get_first_column_from_homogen_table) {
...
+    REQUIRE(col.has_mutable_data());
 
     for (std::int64_t i = 0; i < col.get_count(); i++) {
-        ASSERT_FLOAT_EQ(col[i], t.get_data<float>()[i * t.get_column_count()]);
+        REQUIRE(col[i] == t.get_data<float>()[i * t.get_column_count()]);
     }
 }
 
-TEST(column_accessor_test, can_get_second_column_from_homogen_table_with_conversion) {
+TEST("Column accessor host test: Can get second column from homogen_table with conversion") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:43]:
```suggestion
TEST("Column accessor host test: Can get the second column from homogen_table with conversion") {
```

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 2]
```diff
@@ -50,15 +49,15 @@ TEST(column_accessor_test, can_get_second_column_from_homogen_table_with_convers
...
+    REQUIRE(col.has_mutable_data());
 
     for (std::int64_t i = 0; i < col.get_count(); i++) {
-        ASSERT_DOUBLE_EQ(col[i], double(t.get_data<float>()[i * t.get_column_count() + 1]));
+        REQUIRE(col[i] == double(t.get_data<float>()[i * t.get_column_count() + 1]));
     }
 }
 
-TEST(column_accessor_test, can_get_first_column_from_homogen_table_with_subset_of_rows) {
+TEST("Column accessor host test: Can get first column from homogen_table with subset of rows") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:60]:
```suggestion
TEST("Column accessor host test: Can get the first column from homogen_table with subset of rows") {
```

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 3]
```diff
@@ -101,57 +100,58 @@ TEST(column_accessor_test, can_get_columns_from_homogen_table_builder) {
...
-    ASSERT_THROW(acc.push(column_data, 0, { 3, 4 }), dal::range_error);
-    ASSERT_THROW(acc.push(column_data, 2, { 1, 2 }), dal::range_error);
+    REQUIRE_THROWS_AS(acc.push(column_data, 0, { 0, 2 }), dal::range_error);
+    REQUIRE_THROWS_AS(acc.push(column_data, 0, { 3, 4 }), dal::range_error);
+    REQUIRE_THROWS_AS(acc.push(column_data, 2, { 1, 2 }), dal::range_error);
 }
 
 #ifdef ONEDAL_DATA_PARALLEL
-TEST(column_accessor_dpc_test, can_get_first_column_from_homogen_table) {
+TEST("Column accessor dpc test: Can get first column from homogen_table") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:154]:
```suggestion
TEST("Column accessor DPC test: Can get the first column from homogen_table") {
```

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 4]
```diff
@@ -166,16 +166,16 @@ TEST(column_accessor_dpc_test, can_get_first_column_from_homogen_table) {
...
+    REQUIRE(sycl::get_pointer_type(col.get_data(), q.get_context()) == sycl::usm::alloc::shared);
 
     for (std::int64_t i = 0; i < col.get_count(); i++) {
-        ASSERT_FLOAT_EQ(col[i], i * 2 + 1);
+        REQUIRE(col[i] == i * 2 + 1);
     }
 }
 
-TEST(column_accessor_dpc_test, can_get_second_column_from_homogen_table_with_conversion) {
+TEST("Column accessor dpc test: Can get second column from homogen_table with conversion") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:178]:
```suggestion
TEST("Column accessor DPC test: Can get the second column from homogen_table with conversion") {
```

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 5]
```diff
@@ -190,16 +190,16 @@ TEST(column_accessor_dpc_test, can_get_second_column_from_homogen_table_with_con
...
+    REQUIRE(sycl::get_pointer_type(col.get_data(), q.get_context()) == sycl::usm::alloc::shared);
 
     for (std::int64_t i = 0; i < col.get_count(); i++) {
-        ASSERT_EQ(col[i], i * 2 + 2);
+        REQUIRE(col[i] == i * 2 + 2);
     }
 }
 
-TEST(column_accessor_dpc_test, can_get_first_column_from_homogen_table_with_subset_of_rows) {
+TEST("Column accessor dpc test: Can get first column from homogen_table with subset of rows") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:202]:
```suggestion
TEST("Column accessor DPC test: Can get the first column from homogen_table with subset of rows") {
```

File: .ci/pipeline/ci.yml [hunk 6]
```diff
@@ -156,24 +156,18 @@ jobs:
...
-      sudo apt-get install gcc-multilib \
-                           g++-multilib \
-                           binutils \
-                           openjdk-11-jdk \
-                           $(BAZEL_VERSION)
+      # Download bazelisk
+      export SHA256="168851e70cf5f95c0e215e7f3aaca5132ffc3c8dd8f585a4157b0be2b53cfe32 bazelisk-linux-amd64"
+      wget https://github.com/bazelbuild/bazelisk/releases/download/v1.16.0/bazelisk-linux-amd64
+      echo ${SHA256} | sha256sum --check
+      # "Install" bazelisk
```
Comment (alexandraepan) [inline .ci/pipeline/ci.yml:168]:
```suggestion
      # Install Bazelisk
```

Comment (alexandraepan) [reply .ci/pipeline/ci.yml:168]: maybe Configure Bazelisk?

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 7]
```diff
@@ -14,14 +14,13 @@
...
-#include <gtest/gtest.h>
-
 #include "oneapi/dal/table/column_accessor.hpp"
+#include "oneapi/dal/test/engine/common.hpp"
 #include "oneapi/dal/table/homogen.hpp"
 
 namespace oneapi::dal {
 
-TEST(column_accessor_test, can_get_first_column_from_homogen_table) {
+TEST("Column accessor host test: Can get first column from homogen_table") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:23]:
```suggestion
TEST("Column accessor host test: Can get the first column from homogen_table") {
```

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 8]
```diff
@@ -101,57 +100,58 @@ TEST(column_accessor_test, can_get_columns_from_homogen_table_builder) {
...
 
-    ASSERT_FLOAT_EQ(col[0], 5.f);
-    ASSERT_FLOAT_EQ(col[1], 6.f);
+    REQUIRE(col[0] == 5.f);
+    REQUIRE(col[1] == 6.f);
 }
 
-TEST(column_accessor_test, can_get_column_values_from_column_major_homogen_table_with_conversion) {
+TEST(
+    "Column accessor host test: Can get column values from column major homogen_table with conversion") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:126]:
```suggestion
    "Column accessor host test: Can get column values from major column homogen_table with conversion") {
```

File: cpp/oneapi/dal/table/test/column_accessor.cpp [hunk 9]
```diff
@@ -313,28 +313,28 @@ TEST(column_accessor_dpc_test,
...
-    ASSERT_EQ(col[0], 5);
-    ASSERT_EQ(col[1], 6);
+    REQUIRE(col[0] == 5);
+    REQUIRE(col[1] == 6);
 
     sycl::free(data, q);
 }
 
-TEST(column_accessor_bad_arg_dpc_test, invalid_range) {
+TEST("Column accessor dpc test: Invalid range") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/column_accessor.cpp:324]:
```suggestion
TEST("Column accessor DPC test: Invalid range") {
```

File: cpp/oneapi/dal/table/test/common.cpp [hunk 10]
```diff
@@ -15,39 +15,39 @@
...
 
 #include "oneapi/dal/table/common.hpp"
-#include "gtest/gtest.h"
+#include "oneapi/dal/test/engine/common.hpp"
 
 using namespace oneapi::dal;
 using namespace oneapi;
 
-TEST(table_test, can_construct_empty_table) {
+TEST("Table test: Can construct empty table") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/common.cpp:23]:
```suggestion
TEST("Table test: Can construct an empty table") {
```

File: cpp/oneapi/dal/table/test/common.cpp [hunk 11]
```diff
@@ -15,39 +15,39 @@
...
-    ASSERT_THROW(md.get_feature_type(-1), dal::out_of_range);
-    ASSERT_THROW(md.get_feature_type(n_features), dal::out_of_range);
-    ASSERT_NO_THROW(md.get_feature_type(0));
+    REQUIRE_THROWS_AS(md.get_feature_type(-1), dal::out_of_range);
+    REQUIRE_THROWS_AS(md.get_feature_type(n_features), dal::out_of_range);
+    REQUIRE_NOTHROW(md.get_feature_type(0));
 }
 
-TEST(simple_metadata_bad_arg_tests, get_data_type) {
+TEST("Simple metadata test: Get data type") {
```
Comment (alexandraepan) [inline cpp/oneapi/dal/table/test/common.cpp:43]:
```suggestion
TEST("Simple metadata test: Get a data type") {
```

File: .ci/pipeline/ci.yml (hunk 6 above)
Comment (KulikovNikita) [reply .ci/pipeline/ci.yml:168]: In this particular case it is neither installation nor configuration. Thus the word "install" is in quotes

File: cpp/oneapi/dal/algo/louvain/test/louvain_test.cpp [hunk 12]
```diff
@@ -224,9 +224,12 @@ struct GraphTypeInfo<std::int32_t> {
 
 class louvain_test {
 public:
+    static bool is_nan(double x) {
+        return x != x;
+    }
```
Comment (Alexsandruss) [inline cpp/oneapi/dal/algo/louvain/test/louvain_test.cpp:229]: Are you sure this will work for all nan values?

**Review by Alexsandruss — COMMENTED** [review]
Bazel testing in private CI failed so fix or justification for it is required.

File: cpp/oneapi/dal/algo/louvain/test/louvain_test.cpp (hunk 12 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/algo/louvain/test/louvain_test.cpp:229]: Yes. It works: https://en.wikipedia.org/wiki/NaN#Comparison_with_NaN

**Comment by KulikovNikita** [comment]
Updated: https://intel-ci.intel.com/edfad31c-8f9a-f142-bc12-a4bf010d0e2e

**Review by homksei — APPROVED** [review]

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**